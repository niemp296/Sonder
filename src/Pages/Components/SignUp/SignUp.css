.outerContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: 2%;
    background-color: white;
}

.registrationForm{
    width: 30%;
    height: 60%;
    display: flex;
    flex-direction: column;
    padding: 20px 40px;
    border-radius: 30px;
    box-shadow: 0px 10px 50px rgb(207, 207, 207);
    background-color: white;
    padding-bottom: 2%;
}

.title {
    font-family: 'Baskerville';
    font-size: 5em;
    color: #E58C8A;
    margin-bottom: 7%;
    margin-top: 5%;
    font-weight: lighter;
}

form {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
}

h1 {
    text-align: center;
    width: 100%;
    color: #343434;
    font-weight: bold;
    font-family: 'Poppins';
    font-size: 1em;
}

label {
    font-size: 1em;
    margin-bottom: 0.25em;
    color: #343434;
    font-weight: bold;
    font-size: 14px;
    line-height: 21px;
}

input {
    padding: 10px 10px;
    border-radius: 5px;
    outline: none;
    border: 1px solid #cfcfcf;
}

input::placeholader {
    font-size: 1.2em;
    font-weight: lighter;
    color: #343434;
}

input.error {
    border: 1px solid red;
}

.errorMessage {
    color: red;
    font-size: 0.75em;
    display: relative;
}

.firstName {
    margin-right: 1%;
}

.lastName {
    margin-left: 1%;
}

.firstName,
.lastName,
.areyou,
.email,
.password {
    display: flex;
    flex-direction: column;
    margin-bottom: 15px;
}

.firstName,
.lastName {
    width: 49%;
}

.areyou,
.email,
.password {
    width: 100%;
}

.createAccount {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.createAccount button {
    background-color: #E58C8A;
    color: #fff;
    border: 2px solid #fff;
    width: 100%;
    margin-top: 1em;
    padding: 8px 0px;
    font-size: 1.5em;
    font-weight: bold;
    letter-spacing: 1px;
    margin-bottom: 0.7em;
}

.createAccount small {
    color: #343434;
    font-weight: '600';
    font-size: 18px;
    text-decoration-line: underline;
    text-transform: capitalize;
    font-family: 'Poppins';
    font-style: normal;
}

/* The container */
.container {
    margin-top: 5%;
    display: block;
    position: relative;
    padding-left: 35px;
    margin-bottom: 12px;
    cursor: pointer;
    user-select: none;
    color: #343434;
    font-family: 'Poppins';
    font-size: 1em;
  }
  
  /* Hide the browser's default radio button */
.container input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}
  
  /* Create a custom radio button */
.checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 20px;
    width: 20px;
    background-color: #eee;
    border-radius: 50%;
}
  
  /* On mouse-over, add a grey background color */
.container:hover input ~ .checkmark {
    background-color: #ccc;
}
  
  /* When the radio button is checked, add a blue background */
.container input:checked ~ .checkmark {
    background-color: #2196F3;
}
  
/* Create the indicator (the dot/circle - hidden when not checked) */
.checkmark:after {
content: "";
position: absolute;
display: none;
}

/* Show the indicator (dot/circle) when checked */
.container input:checked ~ .checkmark:after {
display: block;
}

/* Style the indicator (dot/circle) */
.container .checkmark:after {
    top: 6px;
    left: 6px;
    width: 8px;
    height: 8px;
    border-radius: 50%;
    background: white;
}

  