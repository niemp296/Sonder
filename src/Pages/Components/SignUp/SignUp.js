import React, { Component } from "react";
import "./SignUp.css";
import Link from '@material-ui/core/Link';
import Header from '../Header/Header';
import axios from 'axios';
import {Redirect} from "react-router-dom"

const checkEmailFormat = RegExp(
  /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
);

const formValid = ({ formErrors, ...rest }) => {
  let valid = true;

  // validate form errors being empty
  Object.values(formErrors).forEach(val => {
    val.length > 0 && (valid = false);
  });

  // validate the form was filled out
  Object.values(rest).forEach(val => {
    val === null && (valid = false);
  });

  return valid;
};

class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      firstName: null,
      lastName: null,
      traveler: false,
      advertiser: false,
      email: null,
      password: null,
      confirmPassword: null,
      formErrors: {
        firstName: "",
        lastName: "",
        email: "",
        password: "",
        confirmPassword: ""
      },
      isRegistered: false
    };
  }

  handleSubmit = e => {
    e.preventDefault();

    if (formValid(this.state)) {
      if (this.state.firstName !== "") {
        axios.post('http://localhost:5000/api/signUp', this.state)
        .then(response => {
          if(response.data === 400 || response.data === "400"){
              alert("User Already exist");
              console.log("user already exists");
            }
          else if(response.data === 200 || response.data === "200"){
            this.setState({
              isRegistered: true
            })
            alert("You are succesfully registered");
          } 
        })
        .catch(function(error){
          console.log("There's an error passing in data to the back-end")
          console.log(error);
        });
    } else {
        alert("The search query cannot be empty")
    }

    } 
    else {
      console.error("FORM INVALID - DISPLAY ERROR MESSAGE");
    }
  };

  handleChange = e => {
    e.preventDefault();
    const { name, value } = e.target;
    let formErrors = { ...this.state.formErrors };

    switch (name) {
      case "firstName":
        formErrors.firstName =
          value.length < 1 ? "minimum 1 characaters required" : "";
        break;
      case "lastName":
        formErrors.lastName =
          value.length < 1 ? "minimum 1 characaters required" : "";
        break;
      case "username":
        formErrors.username =
          value.length < 3 ? "minimum 3 characaters required" : "";
        break;
      case "email":
        formErrors.email = checkEmailFormat.test(value) ? "" : "invalid email address";
        break;
      case "password":
        formErrors.password =
          value.length < 6 ? "minimum 6 characaters required" : "";
        break;
      case "confirmPassword":	
        formErrors.confirmPassword =	
          (value !== this.state.password) ? "password does not match" : "";	
        break;
      default:
        break;
    }
    this.setState({ formErrors, [name]: value });
  };

  handleRadioChange = e =>{
    const { name } = e.target;
    switch(name){
      case "traveler":
          this.setState({traveler: true, advertiser: false});
          break;
      case "advertiser":
        this.setState({advertiser: true, traveler:false});
        break;
      default:
        break;
    }
    
  }

  render() {
    const { formErrors } = this.state;
    if(this.state.isRegistered){
      return (
        <Redirect to ="/sign-in"></Redirect>
      )
    }
    return (
        <div>
            <Header/>     
        <div className="outerContainer">
            <div className="registrationForm">
                <h1 className="title">
                    Sign Up
                </h1>
            <form action ="http://127.0.0.1:5000/api/signUp" method ="post" onSubmit={this.handleSubmit} noValidate>
                <div className="firstName">
                <label htmlFor="firstName">First Name</label>
                <input
                    className={formErrors.firstName.length > 0 ? "error" : null}
                    placeholder="First Name"
                    type="text"
                    name="firstName"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.firstName.length > 0 && (
                    <span className="errorMessage">{formErrors.firstName}</span>
                )}
                </div>
                <div className="lastName">
                <label htmlFor="lastName">Last Name</label>
                <input
                    className={formErrors.lastName.length > 0 ? "error" : null}
                    placeholder="Last Name"
                    type="text"
                    name="lastName"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.lastName.length > 0 && (
                    <span className="errorMessage">{formErrors.lastName}</span>
                )}
                </div>
                <div>
                {/* TODO: connect state varible traveler and advertiser here  */}
                <label className="areyou">Are you: </label>
                    <label className="container"> Traveler
                        <input 
                            type="radio" 
                            name="traveler"
                            onChange={this.handleRadioChange.bind(this)}
                            checked={this.state.traveler}
                            />
                        <span className="checkmark"></span>
                    </label>
                    <label className="container"> Advertiser
                        <input 
                            type="radio" 
                            name="advertiser"
                            onChange={this.handleRadioChange.bind(this)}
                            checked={this.state.advertiser}
                            />
                        <span className="checkmark"></span>
                    </label>
                </div>
                <div className="email">
                <label htmlFor="email">Email</label>
                <input
                    className={formErrors.email.length > 0 ? "error" : null}
                    placeholder="Email"
                    type="email"
                    name="email"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.email.length > 0 && (
                    <span className="errorMessage">{formErrors.email}</span>
                )}
                </div>
                <div className="password">
                <label htmlFor="password">Password</label>
                <input
                    className={formErrors.password.length > 0 ? "error" : null}
                    placeholder="Password"
                    type="password"
                    name="password"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.password.length > 0 && (
                    <span className="errorMessage">{formErrors.password}</span>
                )}
                </div>
                <div className="password">
                <label htmlFor="password">Confirm Password</label>
                <input
                    className={formErrors.confirmPassword.length > 0 ? "error" : null}
                    placeholder="Confirm Password"
                    type="password"
                    name="confirmPassword"
                    noValidate
                    onChange={this.handleChange}
                />
                {formErrors.confirmPassword.length > 0 && (
                    <span className="errorMessage">{formErrors.confirmPassword}</span>
                )}
                </div>
                <div className="createAccount">
                <button type="submit">Create Account</button>
                <Link href="/sign-in" variant="body">
                    <small>  {"Already have an account? Sign in"} </small>
                </Link>
                </div>
            </form>
            </div>
        </div>
      </div>
    );
  }
}

export default SignUp;