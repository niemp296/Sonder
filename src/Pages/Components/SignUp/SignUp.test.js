import React from 'react';
import { mount, shallow , configure} from 'enzyme';
import SignUp from './SignUp';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe("Test case for SignUp layout", () => {
    it("should render 5 text inpubox, 2 radio input box, 1 button", () => {
        const wrapper = shallow(<SignUp />);
        expect(wrapper.find('input').length).toEqual(7);
        expect(wrapper.find('input[type="radio"]').length).toEqual(2);
        expect(wrapper.find('button').length).toEqual(1);
    });
});

describe('Input box is working', () =>{
    let wrapper = mount(<SignUp />);
    const preventDefault = jest.fn();

    it('input first name', () =>{
        wrapper.find('input[name="firstName"]').simulate('change', {target: {name: 'firstName', value: 'arandomfirstname'}});
        expect(wrapper.state('firstName')).toEqual('arandomfirstname');
    })
    it('input last name', () =>{
        let lastName= "arandomlastname";
        wrapper.find('input[name="lastName"]').simulate('change', {target: {name: 'lastName', value: lastName}});
        expect(wrapper.state('lastName')).toEqual(lastName);
    })
    it('input email', () =>{
        let email= "arandomemail@randomdomain.com";
        wrapper.find('input[name="password"]').simulate('change', {target: {name: 'email', value: email}});
        expect(wrapper.state('email')).toEqual(email);
    })
    it('input password', () =>{
        let password= "$up3rR4nd0mM3g4P4ssw0rd5";
        wrapper.find('input[name="password"]').simulate('change', {target: {name: 'password', value: password}});
        expect(wrapper.state('password')).toEqual(password);
    })
    it('input confirmPassword', () =>{
        let password= "$up3rR4nd0mM3g4P4ssw0rd5";
        wrapper.find('input[name="confirmPassword"]').simulate('change', {target: {name: 'confirmPassword', value: password}});
        expect(wrapper.state('confirmPassword')).toEqual(password);
    })

    it('check traveller radio', () =>{
        wrapper.find('input[name="traveler"]').simulate('change');
        expect(wrapper.state('traveler')).toEqual(true);
        expect(wrapper.state('advertiser')).toEqual(false);
    })
    it('check advertiser radio', () =>{
        wrapper.find('input[name="advertiser"]').simulate('change');
        expect(wrapper.state('traveler')).toEqual(false);
        expect(wrapper.state('advertiser')).toEqual(true);
    })
})
