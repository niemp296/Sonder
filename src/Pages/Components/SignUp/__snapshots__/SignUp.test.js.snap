// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for SignIn layout should render initial layout 1`] = `
Array [
  <div>
    <Header />
    <div
      className="outerContainer"
    >
      <div
        className="registrationForm"
      >
        <h1
          className="title"
        >
          Sign Up
        </h1>
        <form
          action="http://127.0.0.1:5000/api/signUp"
          method="post"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="firstName"
          >
            <label
              htmlFor="firstName"
            >
              First Name
            </label>
            <input
              className={null}
              name="firstName"
              noValidate={true}
              onChange={[Function]}
              placeholder="First Name"
              type="text"
            />
          </div>
          <div
            className="lastName"
          >
            <label
              htmlFor="lastName"
            >
              Last Name
            </label>
            <input
              className={null}
              name="lastName"
              noValidate={true}
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
            />
          </div>
          <div>
            <label
              className="areyou"
            >
              Are you: 
            </label>
            <label
              className="container"
            >
               Traveler
              <input
                checked={false}
                name="traveler"
                onChange={[Function]}
                type="radio"
              />
              <span
                className="checkmark"
              />
            </label>
            <label
              className="container"
            >
               Advertiser
              <input
                checked={false}
                name="advertiser"
                onChange={[Function]}
                type="radio"
              />
              <span
                className="checkmark"
              />
            </label>
          </div>
          <div
            className="email"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className={null}
              name="email"
              noValidate={true}
              onChange={[Function]}
              placeholder="Email"
              type="email"
            />
          </div>
          <div
            className="password"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className={null}
              name="password"
              noValidate={true}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="password"
          >
            <label
              htmlFor="password"
            >
              Confirm Password
            </label>
            <input
              className={null}
              name="confirmPassword"
              noValidate={true}
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
            />
          </div>
          <div
            className="createAccount"
          >
            <button
              type="submit"
            >
              Create Account
            </button>
            <ForwardRef(WithStyles)
              href="/sign-in"
              variant="body"
            >
              <small>
                  
                Already have an account? Sign in
                 
              </small>
            </ForwardRef(WithStyles)>
          </div>
        </form>
      </div>
    </div>
  </div>,
]
`;
